mlx_graphs/nn/message_passing.py

```diff
@@ -25,27 +28,37 @@ def __init__(self, aggr="add"):
         super().__init__()
 
         self.aggr = aggr
+        self.node_dim = None
 
     def __call__(self, x: mx.array, edge_index: mx.array, **kwargs: Any):
         raise NotImplementedError
 
-    def propagate(self, x: mx.array, edge_index: mx.array, **kwargs: Any) -> mx.array:
+    def propagate(self, x: Union[mx.array, ArrayTuple], edge_index: mx.array, **kwargs: Any) -> mx.array:
```

Comment:
I'm not sure about the utility of the `ArrayTuple`, probably better to stick to `mx.array`

Reply:
Me too, I'll remove it now and we'll see in the near future if some long typings are frequently used in the project